crates/sparrow-physical/src/step.rs

```diff
@@ -0,0 +1,59 @@
+use arrow_schema::SchemaRef;
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+#[repr(transparent)]
+#[serde(transparent)]
+pub struct StepId(usize);
+
+/// A single step in the physical plan.
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub struct Step {
+    pub id: StepId,
+    /// The kind of step being performed.
+    pub kind: StepKind,
+    /// Steps which act as input (children) of this step.
+    pub children: Vec<StepId>,
```

Comment:
I am not 100% sure why we prefer `children` rather than `input`, `upstream` or some other word that does not denote parent-child relationship.

Reply:
I think we adopted it from `egg`. I'm OK calling this `inputs` (in `Step`) and maybe `args` (in `Expression`).